prime-router/src/main/kotlin/azure/observability/event/ReportRouteEvent.kt

```diff
@@ -14,4 +14,27 @@ data class ReportRouteEvent(
     val receiver: String?,
     val observations: List<ObservationSummary>,
     val bundleSize: Int,
+) : AzureCustomEvent
+
+data class DestinationFilterReportRoutedEvent(
```

Comment:
Still not sure on names, but `ReceiverSelectedEvent` was floating in my head after we met